### PR TITLE
Add Code Quailty chapter

### PR DESCRIPTION
**Pull request by andreicek**
This pull request should specify how we create pull requests and how we review them.

I've also moved the existing chapter to a Tools page and updated some links.

### CODE REVIEW

File: Code Quality/Creating a pull request.md
Comment (andreicek) [reply Code Quality/Creating a pull request.md]: Yup, 37bb9d5

File: Code Quality/Tools.md
Comment (DarkoKukovec) [inline Code Quality/Tools.md:163]: Since you're already updating this chapter, could you also fix the links? Darac discovered yesterday that they don't work.

File: Code Quality/Reviewing a pull request.md
Comment (DarkoKukovec) [inline Code Quality/Reviewing a pull request.md]: Also worth mentioning that it should be taken into account during estimates?

File: Code Quality/Creating a pull request.md
Comment (fvoska) [inline Code Quality/Creating a pull request.md:14]: `screen shoot` 😂

File: Code Quality/Creating a pull request.md
Comment (fvoska) [inline Code Quality/Creating a pull request.md:29]: `screen shoot` 🔫 🔫 🔫

File: Code Quality/Creating a pull request.md
Comment (fvoska) [inline Code Quality/Creating a pull request.md:18]: I found many conflicting sources about whether bullet lists should begin with a capital letter or not and how they should be punctuated. It seems to vary in different cases and is partially up to the author to decide. Whatever the case, I think we should at least be consistent. The bullet list above has a mix of upper an lower case, this list is lower case and the list below is upper case.

File: Code Quality/Reviewing a pull request.md
Comment (DarkoKukovec) [inline Code Quality/Reviewing a pull request.md:60]: Worth investigating tools that would verify this? I've seen bots and GH actions for this purpose.

File: Code Quality/Reviewing a pull request.md
Comment (DarkoKukovec) [inline Code Quality/Reviewing a pull request.md:58]: This statement is correct, however, should this be the focus of a PR review? Since we're already having tools for formatting, this seems like a distraction to the reviewer.

File: Code Quality/Reviewing a pull request.md
Comment (fvoska) [inline Code Quality/Reviewing a pull request.md]: Scharfes S is probably a typo

File: Code Quality/Tools.md
Comment (fvoska) [inline Code Quality/Tools.md:163]: Are you sure this change will not mess up rendering when deployed? I know that our markdown-to-html is a bit fussy about indentations and lists. Can you please merge to staging and check?

**Review by fvoska — CHANGES_REQUESTED** [review]

File: Code Quality/Creating a pull request.md
Comment (andreicek) [reply Code Quality/Creating a pull request.md]: Fixed.

File: Code Quality/Creating a pull request.md
Comment (andreicek) [reply Code Quality/Creating a pull request.md]: Fixed.

File: Code Quality/Reviewing a pull request.md
Comment (andreicek) [reply Code Quality/Reviewing a pull request.md]: Fixed. But it was not!

File: Code Quality/Reviewing a pull request.md
Comment (fvoska) [reply Code Quality/Reviewing a pull request.md:60]: If using karma, you can set up coverage gates so that the build for the branch will fail if the coverage drops below some threshold and this will block merging.

I am not sure if there is a way to check that the coverage dropped compared to target branch, but not low enough to trigger the gate (e.g. master has 82% coverage, gate is 80% and PR drops it to 81%). Gate should also be increased from time to time

File: Code Quality/Reviewing a pull request.md
Comment (andreicek) [reply Code Quality/Reviewing a pull request.md:58]: I agree with you 100%. This should probably go to the creating a PR document instead. Intention with this was "do not review if that's not true".

I'll reword and place this in the creating document as well.

File: Code Quality/Reviewing a pull request.md
Comment (DarkoKukovec) [reply Code Quality/Reviewing a pull request.md:60]: I know about this, jest also has it. The question is if we want to have a fixed gate or an increasing gate

File: Code Quality/Reviewing a pull request.md
Comment (andreicek) [reply Code Quality/Reviewing a pull request.md]: HM - Good point!

File: Code Quality/Creating a pull request.md
Comment (andreicek) [reply Code Quality/Creating a pull request.md]: Fixed.

File: Code Quality/Tools.md
Comment (andreicek) [reply Code Quality/Tools.md:163]: I'll check!

Comment (andreicek) [reply Code Quality/Tools.md]: Do we not have eslint setup for angular?

File: Code Quality/Reviewing a pull request.md
Comment (safo6m) [inline Code Quality/Reviewing a pull request.md:40]: This part is weird. You are talking about assigning a certain part of the day for reviewing pull requests and then in the second part you state that this should be done "first thing tomorrow".

I would go in a different direction. When and how often pull requests should be reviewed is up to you and the project team but as a rule of thumb - a pull request shouldn't be unchecked for more than 24 hours. That means that you should review pull requests at least once a day.
Urgent fixes should, of course, be checked as soon as possible.

Comment (andreicek) [reply Code Quality/Reviewing a pull request.md:40]: Yeah I agree - this was the idea; and the suggestion was to have as a fixed portion of the day (no matter if it's morning or evening or whatever).

I'll reword this a bit.

File: Code Quality/Creating a pull request.md
Comment (andreicek) [reply Code Quality/Creating a pull request.md]: Done 758a44f

File: Code Quality/Reviewing a pull request.md
Comment (andreicek) [reply Code Quality/Reviewing a pull request.md]: I cleared it up a bit. Care to give it another read? 758a44f

File: Code Quality/Reviewing a pull request.md
Comment (mamihelj09) [inline Code Quality/Reviewing a pull request.md:22]: for this example I think we need an explanation why would it be better to move that file, like 
```
> this file should go to `components/shared` because this is atom component and can be reused on another page
```

Or something like that. If we give an explanation it might open a discussion about it.

Comment (andreicek) [reply Code Quality/Reviewing a pull request.md:22]: Amazing thanks

**Review by DarkoKukovec — APPROVED** [review]

File: Code Quality/Tools.md
Comment (andreicek) [reply Code Quality/Tools.md]: Works. This is how prettier formats it.

**Review by mamihelj09 — APPROVED** [review]

**Review by badurinantun — APPROVED** [review]

File: Code Quality/Reviewing a pull request.md
Comment (safo6m) [reply Code Quality/Reviewing a pull request.md]: Great, I think you cleared everything up now 👍

**Review by safo6m — APPROVED** [review]

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:15]:
```suggestion
-   Be open for discussion, but do not bikeshed
```
https://en.wiktionary.org/wiki/bikeshedding

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:22]:
```suggestion
> this file should go to `components/shared` because this is an atom component and can be reused on another page
```

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:54]:
```suggestion
You should have the project you're reviewing checked out locally and you should run **both** the project and the tests on **every** pull request. Your review is not complete unless you click through the feature and verify at least the happy path.
```
image

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:56]:
```suggestion
Take a look at the code coverage and look for missing things there, are all branches tested or even developed?
```

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:68]:
```suggestion
Some of these checks can be automated on pull requests (e.g. code coverage). Talk to your resident devops to help you set this up.
```

File: Code Quality/Reviewing a pull request.md
Comment (kristijanbambir) [inline Code Quality/Reviewing a pull request.md:72]:
```suggestion
Every developer is responsible for their code. But - you as a reviewer share the responsibility as well when you sign off a feature and allow it to merge.
```

**Review by kristijanbambir — CHANGES_REQUESTED** [review]

**Review by kristijanbambir — APPROVED** [review]

**Review by fvoska — APPROVED** [review]

**Pull request merged**
